docs/profiling.md

```diff
@@ -1,7 +1,128 @@
+   docs](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras#use_the_tensorflow_profiler_to_profile_model_training_performance)
+   for more details on using the trace viewer.<br /><br />
+
+1. By default, the events in the trace viewer are mostly low-level internal JAX
+   functions. You can add your own events and functions by using
+   {func}`jax.profiler.TraceContext` and {func}`jax.profiler.trace_function` in
+   your code and capturing a new profile.
+
+### Troubleshooting
+
+#### GPU profiling
+
+Programs running on GPU should produce traces for the GPU streams near the top
+of the trace viewer. If you're only seeing the host traces, check your program
+logs and/or output for the following error messages.
+
+**If you get an error like: `Could not load dynamic library 'libcupti.so.10.1'`**<br />
+Full error:
+```
+W external/org_tensorflow/tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcupti.so.10.1'; dlerror: libcupti.so.10.1: cannot open shared object file: No such file or directory
+2020-06-12 13:19:59.822799: E external/org_tensorflow/tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1422] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error CUPTI could not be loaded or symbol could not be found.
+```
+
+Add the path to `libcupti.so` to the environment variable `LD_LIBRARY_PATH`.
+(Try `locate libcupti.so` to find the path.) For example:
+```shell
+export LD_LIBRARY_PATH=/usr/local/cuda-10.1/extras/CUPTI/lib64/:$LD_LIBRARY_PATH
+```
+
+**If you get an error like: `failed with error CUPTI_ERROR_INSUFFICIENT_PRIVILEGES`**<br />
+Full error:
+```shell
+E external/org_tensorflow/tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1445] function cupti_interface_->EnableCallback( 0 , subscriber_, CUPTI_CB_DOMAIN_DRIVER_API, cbid)failed with error CUPTI_ERROR_INSUFFICIENT_PRIVILEGES
+2020-06-12 14:31:54.097791: E external/org_tensorflow/tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1487] function cupti_interface_->ActivityDisable(activity)failed with error CUPTI_ERROR_NOT_INITIALIZED
+```
+
+Run the following commands (note this requires a reboot):
+```shell
+echo 'options nvidia "NVreg_RestrictProfilingToAdminUsers=0"' | sudo tee -a /etc/modprobe.d/nvidia-kernel-common.conf
+sudo update-initramfs -u
+sudo reboot now
+```
+
+(remote_profiling)=
+#### Profiling on a remote machine
+
+If the JAX program you'd like to profile is running on a remote machine, one
+option is to run all the instructions above on the remote machine (in
+particular, start the TensorBoard server on the remote machine), then use SSH
+local port forwarding to access the TensorBoard web UI from your local
+machine. Use the following SSH command to forward the default TensorBoard port
+6006 from the local to the remote machine:
+
+```shell
+ssh -L 6006:localhost:6006 <remote server address>
+```
 
 ## nvprof
```

Comment:
I think nvprof has been deprecated and replaced by something called NSight, although I've never used it.

Reply:
I changed nvprof -> Nsight (I thought about just removing this section, but maybe it'll inspire someone to try it).